### ISSUE TITLE
Fix S3900: Rule should not raise an issue when compared with IsNullOrWhitespace/IsNullOrEmpty

### ISSUE THREAD

**Issue by manchas** [SonarSource/sonar-csharp#693]
### Description

The rule gives a false positive, indicating that a string variable hasn't been tested for null befor its use. BUT the variable is tested, just not with an `x == null` but with a `string.IsNullOrWhiteSpace()` or with a `string.IsNullOrEmpty()`. 

### Repro steps
```
public void a(string b)
{
     if (string.IsNullOrWhiteSpace(b))  return;
     var c = b.Split('_');
     Console.WriteLine(c.Count());
}
```
### Expected behavior

It souldn't fire the rule because the string b is being checked for null before it's use.

### Actual behavior

False positive is reported on line `var c = b.Split('_');`, marking b as being used without previos check.

### Known workarounds

First check for null explicitly as in `if (b == null) return;` and THEN check for IsNullOrWhiteSpace, which effectively beats the whole purpose of the nice string function IsNullOrWhiteSpace

### Related information

* SonarC# Version: SonarLint for Visual Studio 3.4.0.1732
* Visual Studio Version : 15.2 (26430.16)
* If running through the Scanner for MSBuild, its version: N/A

**Comment by Evangelink** [comment]
Hi @manchas ,
Thanks for the feedback. This might be quite tricky for us to solve as we really on Symbolic Execution and we don't support cross-procedural analysis.
@valhristov Would it be hard to add some "known" methods?
